### ISSUE TITLE
Something is stopping BepInEx/Doorstop to load correctly

### ISSUE THREAD

**Issue by ArsArcanumb** [BepInEx/BepInEx#129]
**Describe the bug**

Pretty much the exact same problem as #86 only instead of Koikatsu I'm having trouble with HS2 and AIS. Unfortunately none of the methods in the thread seem to work for me. First of all, I also used Sophos however everything worked fine until a few months ago. This morning I uninstalled Sophos through the same means as #86 and also multiple scripts that are supposed to completely wipe everything Sophos off the PC but the problem remains. So I'm unsure if Sophos really is the reason here. Even weirder is, that Koikatsu RX5.1 works fine while AIS R6 and HS2 R3.1 don't open any window at all. Based on the other thread I don't think doorstop is to blame, it's probably something inside the preloader. Setting ApplyRuntimePatches = false doesn't do anything. I'd just really want to know which program or service is stopping BepInEx. If there is any way to debug this, I'd like to know how.

**To Reproduce**
A short description of what you did for the bug to happen.

No idea. I didn't play too frequently to decide excactly when it happened.

**Expected behavior**

The game to start or at least throw an exception.

**Screenshots and logs**
Include a console log (`output_log.txt` and/or `preloader_xxx_xxx.log`), if the error is visible there.  
If not possible, include a screenshot of the error.

It doesn't get far enough to produce a log. If there is another exe that produces doorstep or preloader logs, I'd be happy to try it out.

**Desktop (please complete the following information):**
 - Game and game version AI-Shoujo BetterRepack R6 and Honey Select 2 Better Repack R3.1: 
 - BepInEx version [e.g. 4.0]: BepInEx v5.2

**Additional context**

I really tried out all the solutions from #86 but I'm at my wits end now.

**Comment by denikson** [comment]
Greetings!

Since Koikatsu works well, as you tell, my guess is that it could potentially be something else. The best option here would be to test with a fresh install of the game if possible. If you have followed the debugging steps outlined in #86 and have replaced Doorstop and other files already, you might want to reinstall the pack anyway as all the debug DLLs in that issue are pretty old and likely incompatible with BepInEx 5.2. Otherwise, try and restore your game to before you tried any DLLs in that issue.

If you still want to proceed with debugging (or you don't have the ability to easily reinstall the game), you can try gathering some Doorstop logs. Download the following ZIP:
[doorstop_with_logging.zip](https://github.com/BepInEx/BepInEx/files/5030984/doorstop_with_logging.zip)
And extract `winhttp.dll` into AI/HS2 folder replacing the one you have.

Next, run the game. That should generate `doorstop_xxxxxxxx.log` in the game folder (where `xxxxxxxx` is a sequence of characters). Send over that log here so we can take a look.

**Comment by ArsArcanumb** [comment]
Hi, thanks for helping me :)

I've already reinstalled multiple times, even older repacks that previously worked without problems but don't anymore.
Here's the log file

> Doorstop started!
EXE Path: D:\ #######\Backups\Spiele\Honey Select 2\HoneySelect2.exe
App dir: D:\ #######\Backups\Spiele\Honey Select 2
Working dir: D:\ #######\Backups\Spiele\Honey Select 2
STDOUT handle at 0000000000000000
Pointer to GetFinalPathNameByHandleA 00007FFAC9944D60
STDOUT handle path: 
DLL Path: D:\ #######\Backups\Spiele\Honey Select 2\WINHTTP.dll
Doorstop DLL Name: WINHTTP
Looking for original DLL from D:\ #######\Backups\Spiele\Honey Select 2\WINHTTP_alt.dll
Looking for original DLL from C:\WINDOWS\system32\WINHTTP.dll
Config; Target assembly: D:\ #######\Backups\Spiele\Honey Select 2\BepInEx\core\BepInEx.Preloader.dll
Redirected output log!
CMDLine: "D:\ #######\Backups\Spiele\Honey Select 2\HoneySelect2.exe"  -logFile "D:\ #######\Backups\Spiele\Honey Select 2\output_log.txt"
Doorstop enabled!
Installing IAT hook
Hook installed!!
Got mono.dll at 00007FFA0ECC0000
Loaded all mono.dll functions
Starting mono domain
STDOUT handle at 0000000000000000
STDOUT handle path: 
Setting config paths: base dir: D:\ #######\Backups\Spiele\Honey Select 2\; config path: D:\ #######\Backups\Spiele\Honey Select 2\HoneySelect2.exe.config
Assembly dir: D:\ #######\Backups\Spiele\Honey Select 2\HoneySelect2_Data\Managed
Loading assembly: D:\ #######\Backups\Spiele\Honey Select 2\BepInEx\core\BepInEx.Preloader.dll
Invoking method!

I've censored out my name after the D:\ directory, but it doesn't contain any special characters so that shouldn't be an issue.
The AIS one is basically the same.

Thank you

**Comment by denikson** [comment]
From the looks of it Doorstop is indeed working fine, so the problem isn't the same as in #86 

To me it seems like you might have broken BepInEx install. I'd try to

* make sure that the path to the game is not too long, as it can cause some issues with loading
* reinstall BepInEx by first just downloading latest [BepInEx version](https://github.com/BepInEx/BepInEx/releases/download/v5.2/BepInEx_x64_5.2.0.0.zip) and extracting it into the game path. Overwrite when asked
* reinstall the whole pack, as to me it seems like something might be wrong with the whole install

**Comment by ArsArcanumb** [comment]
Unfortunately I already tried all of that. Even the discord.
Is there a possibility to debug the preloader? Or any windows application to monitor which programm is stopping the preloader?
It's not that urgent though, maybe if I'm lucky it will be fixed with the next release.

**Comment by ArsArcanumb** [comment]
I got it.
After seeing in Process Monitor that the last thing the exe did was interacting with the timezone registry, I tried setting my timezone to Japanese and suddenly it worked. It's even still working after setting it back.
I have absolutely no clue what changed or why that worked. Maybe some Windows problem.
Well, all is fine now. Thanks for the help and sorry for wasting any of your time.
Have a nice weekend
